### ISSUE TITLE
Garbage printed by escape

### ISSUE THREAD

**Issue by tmccombs** [neovim/neovim#6067]
- `nvim --version`: 
```
NVIM 0.2.0-dev
Build type: RelWithDebInfo
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector --param ssp-buffer-size=4 -DINCLUDE_GENERATED_DECLARATIONS -DHAVE_CONFIG_H -D_GNU_SOURCE -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/build/config -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/src -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/.deps/usr/include -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/.deps/usr/include -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/.deps/usr/include -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/.deps/usr/include -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/.deps/usr/include -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/.deps/usr/include -I/usr/include -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/build/src/nvim/auto -I/build/neovim-RKxPSU/neovim-0.1.7ubuntu1+git201702061523+3114+23~ubuntu14.04.1/build/include
Compiled by root@lgw01-54
```
- Vim (version: 7.4) behaves differently? Yes
- Operating system/version: Ubuntu 14.04
- Terminal name/version: Terminator and xfce4-terminal (but not gnome-terminal)
- `$TERM`: xterm

### Actual behaviour
Escape draws garbage characters on the screen. When I press escape to return to normal mode, a unix block containing "001B" (the hex code for escape) is drawn at the beginning of the status line. Some other commands (such as `%`) will draw garbage (usually that starts with the 001B block) in the editing area. The garbage seems to stay there until I close nvim.

### Expected behaviour
No having garbage on my screen.

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
i{<RETURN>
test<RETURN>
}
<ESC>
%

```

**Comment by tmccombs** [comment]
This doesn't happen in xterm either.

**Comment by justinmk** [comment]
See https://github.com/neovim/neovim/issues/4867 and `man nvim`. Add

    let $NVIM_TUI_ENABLE_CURSOR_SHAPE = 0

to your init.vim, then restart nvim. 

Does that fix it?

**Comment by tmccombs** [comment]
Yes, that fixes it thank you.

**Comment by tmccombs** [comment]
This doesn't work on nvim 0.1.7.

**Comment by justinmk** [comment]
In 0.1.7 don't define NVIM_TUI_ENABLE_CURSOR_SHAPE.

In 0.2, set guicursor to empty:

    :set guicursor=

At any rate, what does 0.1.7 have to do with bugs that have already been fixed? We can't fix bugs in 0.1.7.

**Comment by tmccombs** [comment]
Sorry, I was just hoping there would be a workaround for 0.1.7 (since that is the stable version) to force guicursor to be disabled. As is neovim is unusable in most terminals Ubuntu 16.04. I think because it has an older version of VTE.

**Comment by jamessan** [comment]
In 0.1.7, `'guicursor'` doesn't affect how the cursor in the TUI.  In 0.1.7, the user has to specifically enable cursor shaping by setting `$NVIM_TUI_ENABLE_CURSOR_SHAPE`, so if you want to fix the problem in 0.1.7, stop setting that environment variable.
